File: java/linkedlist.java
public class Main
{  
    Node node;
    static class Node{
        int data;
        Node next;
        Node (int d){
            data=d;
            next=null;
        }
    }
    static Node insertFirst(Node head,int d){
        Node newnode=new Node(d);
        newnode.next=head;
        return newnode;
    }
    static void insertSearchAfter(Node head,int d,int key){
        Node newnode=new Node(d);
        Node temp=head;
        while(temp.data!=key){temp=temp.next;}
        if(temp.next!=null){newnode.next=temp.next;}
        temp.next=newnode;
        
    }
    
    static void insertLast(Node head,int d){
        Node newnode=new Node(d);
        Node temp=head;
        while(temp.next!=null){temp=temp.next;}
        temp.next=newnode;
        
    }
    static Node deleteFirst(Node head){
        return head.next;
    }
    static void deleteSearchAfter(Node head,int key){
        Node prev=null;
        Node temp=head;
        while(temp.data!=key && temp.next!=null){prev=temp;temp=temp.next;}
        if(temp.data==key){prev.next=temp.next;}
        
        
    }
    static void deleteLast(Node head){
        Node prev=null;
        Node temp=head;
        while(temp.next!=null){prev=temp;temp=temp.next;}
        prev.next=null;
        
    }
    static void print(Node node){
        while(node!=null){
            
            System.out.print(node.data+"->");
            node=node.next;
        }System.out.println("null");
    }
	public static void main(String[] args) {
		Node head=new Node(1);
		head=insertFirst(head,5);
		insertSearchAfter(head,15,5);
		insertLast(head,10);
		insertLast(head,20);
		insertLast(head,30);
		insertLast(head,40);
		print(head);
	        head=deleteFirst(head);
	        deleteLast(head);
		print(head);
		
	}
}

File: c/linkedlist.c
#include <stdio.h>
#include <stdlib.h>
struct Node
{
    int data;
    struct Node *next;
    
}*head;
void show(struct Node *head){
    
    while(head!=NULL){printf("%d->",head->data);head=head->next;}
    
}
struct Node* insertFirst(struct Node *head,int d){
    
    struct Node *newnode=(struct Node *)malloc(sizeof(struct Node));
    newnode->data=d;
    newnode->next=head;
    return newnode;
 
}
void insertSearchAfter(struct Node *head,int d,int key){
    
    if(head==NULL){
        printf("node cant be null");
    }
    else{
        struct Node *newnode=(struct Node *)malloc(sizeof(struct Node));
        newnode->data=d;
        while(head->data!=key){head=head->next;}
        newnode->next=head->next;
        head->next=newnode;
        
        
    }
}


void insertLast(struct Node *head,int d){
    
    if(head==NULL){
        printf("node cant be null");
    }
    else{
        struct Node *newnode=(struct Node *)malloc(sizeof(struct Node));
        newnode->data=d;
        while(head->next!=NULL){head=head->next;}
        head->next=newnode;
        
    }
}
struct Node* deleteFirst(struct Node *head){
    
 
    head=head->next;
    return head;
 
}
void deleteSearchAfter(struct Node *head,int key){
    
    if(head==NULL){
        printf("node cant be null");
    }
    else{
        struct Node *prev=NULL;
    
        while(head->data!=key){prev=head;head=head->next;}
        prev->next=head->next;
        
    }
}
void deleteLast(struct Node *head){
    
    if(head==NULL){
        printf("node cant be null");
    }
    else{
        struct Node* prev=NULL;
        while(head->next!=NULL){prev=head;head=head->next;}
        prev->next=NULL;
        
    }
}
int main()

{   
    //struct Node *head=(struct Node*)malloc(sizeof(struct Node));
    head=insertFirst(head,2);
    
    insertLast(head,12);
    insertLast(head,22);
    insertSearchAfter(head,7,12);
    deleteSearchAfter(head,12);
    head=insertFirst(head,2);
    insertLast(head,12);
    insertLast(head,22);
    deleteLast(head);
    head=deleteFirst(head);
    show(head);

    return 0;
}
